rocketship/dce/main.py

```diff
@@ -0,0 +1,271 @@
+"""
+Main DCE-MRI analysis functions for ROCKETSHIP.
+"""
+
+import os
+import numpy as np
+from typing import Dict, List, Optional, Tuple, Any
+import logging
+
+from ..utils.config import load_dce_preferences
+from ..utils.file_io import ImageLoader, VolumeLoader, FileParser
+from ..utils.image_processing import ImageProcessor, ROIAnalyzer
+from .models import ToftsModel, ExtendedToftsModel, PatlakModel, AIFModel
+
+
+logger = logging.getLogger(__name__)
+
+
+class DCEAnalyzer:
+    """Main DCE-MRI analysis class."""
+    
+    def __init__(self, config_file: Optional[str] = None):
+        """Initialize DCE analyzer with configuration."""
+        self.config = load_dce_preferences(config_file)
+        self.models = {
+            'tofts': ToftsModel(),
+            'extended_tofts': ExtendedToftsModel(),
+            'patlak': PatlakModel()
+        }
+    
+    def analyze_dce_data(self, 
+                        input_files: List[str],
+                        output_dir: str,
+                        aif_roi: Optional[np.ndarray] = None,
+                        model_type: str = 'tofts',
+                        verbose: bool = False) -> Dict[str, Any]:
+        """
+        Run complete DCE-MRI analysis.
+        
+        Args:
+            input_files: List of DCE time series files
```

Comment:
DCE requires multiple input files: the dynamic series, the AIF ROI, the brain mask, and the T1 map.